integration-tests/src/test/java/oracle/weblogic/kubernetes/ItExternalNodePortService.java

```diff
@@ -0,0 +1,416 @@
+             "/u01/oracle/wlserver/server/lib/wlthint3client.jar",
+             Paths.get(RESULTS_ROOT, "wlthint3client.jar")));
+    StringBuffer javacCmd = new StringBuffer("javac -cp ");
+    javacCmd.append(Paths.get(RESULTS_ROOT, "wlthint3client.jar "));
+    javacCmd.append(Paths.get(RESOURCE_DIR, "tunneling", "JmsTestClient.java"));
+    javacCmd.append(Paths.get(" -d "));
+    javacCmd.append(Paths.get(RESULTS_ROOT));
+    logger.info("javac command {0}", javacCmd.toString());
+    ExecResult result = assertDoesNotThrow(
+        () -> exec(new String(javacCmd), true));
+    logger.info("javac returned {0}", result.toString());
+    logger.info("javac returned EXIT value {0}", result.exitValue());
+    assertTrue(result.exitValue() == 0, "Client compilation fails");
+  }
+
+  // Build JMS Client inside the Admin Server Pod
+  private void buildClientOnPod() {
+    String destLocation = "/u01/oracle/JmsTestClient.java";
+    assertDoesNotThrow(() -> copyFileToPod(domainNamespace,
+             adminServerPodName, "weblogic-server",
+             Paths.get(RESOURCE_DIR, "tunneling", "JmsTestClient.java"),
+             Paths.get(destLocation)));
+
+    String jarLocation = "/u01/oracle/wlserver/server/lib/wlthint3client.jar";
+    StringBuffer javacCmd = new StringBuffer("kubectl exec -n ");
+    javacCmd.append(domainNamespace);
+    javacCmd.append(" -it ");
+    javacCmd.append(adminServerPodName);
+    javacCmd.append(" -- /bin/bash -c \"");
+    javacCmd.append("javac -cp ");
+    javacCmd.append(jarLocation);
+    javacCmd.append(" JmsTestClient.java ");
+    javacCmd.append(" \"");
+    logger.info("javac command {0}", javacCmd.toString());
+    ExecResult result = assertDoesNotThrow(
+        () -> exec(new String(javacCmd), true));
+    logger.info("javac returned {0}", result.toString());
+    logger.info("javac returned EXIT value {0}", result.exitValue());
+    assertTrue(result.exitValue() == 0, "Client compilation fails");
+  }
+
+  // Run external standalone JMS Client using wlthint3client.jar in classpath.
+  // The client sends 300 messsage to a Uniform Distributed Queue.
+  // Make sure that each destination get excatly 150 messages each.
+  private static Callable<Boolean> runJmsClient(String javaCmd) {
+    return (()  -> {
+      ExecResult result = assertDoesNotThrow(() -> exec(new String(javaCmd), true));
+      logger.info("java returned {0}", result.toString());
+      logger.info("java returned EXIT value {0}", result.exitValue());
+      return ((result.exitValue() == 0));
+    });
+  }
+
+  @AfterAll
+  public void tearDownAll() {
+    if (System.getenv("SKIP_CLEANUP") == null
+        || (System.getenv("SKIP_CLEANUP") != null
+        && System.getenv("SKIP_CLEANUP").equalsIgnoreCase("false"))) {
+      StringBuffer deployIngress = new StringBuffer("kubectl delete -f ");
+      deployIngress.append(Paths.get(RESULTS_ROOT, "cluster.nodeport.svc.yaml"));
+    }
+  }
+
+  private static void createDomainResource(
```

Comment:
could you use method from CommonMiiTestUtils.java ?

Reply:
In the current usecase, I create a domain resource with a pre-defined ConfigMap which add JMS resources to the domain.